packages/swayze/lib/src/core/intents/selection_intents.dart

```diff
@@ -41,14 +41,51 @@ class ExpandSelectionByBlockIntent extends SwayzeIntent {
   const ExpandSelectionByBlockIntent(this.direction);
 }
 
+/// A [SwayzeIntent] to fill a range based on cells from the source range.
+///
+/// This differs from [FillFromRangeIntent] as we know the target range.
+///
+/// See also:
+/// - [TableBodyGestureDetector] that triggers this intent
+class FillRangeIntent extends SwayzeIntent {
+  final Range2D source;
+
+  final Range2D target;
+
+  const FillRangeIntent({
+    required this.source,
+    required this.target,
+  });
+}
+
+/// A [SwayzeIntent] to fill unknown cells from a given range.
+///
+/// This differs from [FillRangeIntent] as we know don't know the target range.
+///
+/// See also:
+/// - [TableBodyGestureDetector] that triggers this intent
+class FillFromRangeIntent extends SwayzeIntent {
```

Comment:
I feel like these names might be confusing / misleading. The `FillRangeIntent` is the one that receives a Range to fill from, so I think that should be called `FillFromRangeIntent`.

This one could be something like: `FillFromNeighbouringBlockIntent`, `FillFromAdjacenteBlockIntent`. 


Or, if we dont want to assume the logic that will be on the app side: `FillIntoTargetIntent` and `FillIntent` / `FillIntoUnknownIntent`

Reply:
I'm terrible at names. These are great. Done.